src/lib.rs

```diff
@@ -110,7 +171,7 @@ macro_rules! from_impl {
     ($from:ty => $to:ty; $by:ident) => (
         impl<'a> From<$from> for UniCase<$to> {
             fn from(s: $from) -> Self {
-                UniCase(s.$by())
+                UniCase::new(s.$by())
```

Comment:
I suppose the `From` impls could just jump straight to `Encoding::Unicode`, is that what you mean?